frontend/src/components/auth/Profile.jsx

```diff
@@ -1,52 +1,192 @@
 import './Profile.css'
-
 import React from 'react'
 import { signOut } from './auth.js'
 import { useState, useEffect } from 'react'
 import { useAuth } from './auth.jsx'
 import { useNavigate } from 'react-router-dom'
+import Record from '../Record.jsx'
+import TTS from '../TTS.jsx'
+
+const interviewQuestions = [
+    "Can you tell me about a time you worked on a team to complete a project? What was your role, and what did you learn from the experience?",
+    "What is your greatest strength and how does it help you in your work?",
+    "quit"
+    // Add more questions as needed
+];
 
 const Profile = () => {
-    const { currentUser } = useAuth()
-    const [user, setUser] = useState(null)
-    const navigate = useNavigate()
+    const { currentUser } = useAuth();
+    const [user, setUser] = useState(null);
+    const [input, setInput] = useState("");
+    const [messages, setMessages] = useState([]);
+    const [interviewStarted, setInterviewStarted] = useState(false);
+    const [currentQuestionIndex, setCurrentQuestionIndex] = useState(0);
+    const [expectingFollowUp, setExpectingFollowUp] = useState(false);
+    const [lastQuestionCheck, setLastQuestionCheck] = useState("")
+    const [prevIsFollowUp, setPrevIsFollowUp] = useState(false)
+
+    const [isUserTurn, setIsUserTurn] = useState(false);
+    const [isLoading, setIsLoading] = useState(false);
+    const [isTranscribing, setIsTranscribing] = useState(false);
+    const [isInterviewOver, setIsInterviewOver] = useState(false);
+    const [spokenMessages, setSpokenMessages] = useState([]);
+    const [newMessages, setNewMessages] = useState([]);
+
+    const navigate = useNavigate();
 
     useEffect(() => {
         if (currentUser) {
             setUser({
                 email: currentUser.email,
                 name: currentUser.displayName || "User",
-            })
+            });
         }
-    }, [currentUser])
+    }, [currentUser]);
 
+    useEffect(() => {
+        const unspokenMessages = messages.filter(msg =>
+            msg.role === "bot" && !spokenMessages.includes(msg.content)
+        );
+        setNewMessages(unspokenMessages);
+    }, [messages, spokenMessages]);
 
     const handleSignout = async () => {
         try {
-            await signOut()
-            navigate('/')
-        }
-        catch (error) {
-            console.error('Error signing out:', error)
+            await signOut();
+            navigate('/');
+        } catch (error) {
+            console.error('Error signing out', error);
         }
+    };
+
+    const handleTranscription = (transcribedText) => {
+        setIsTranscribing(false);
+        setInput(transcribedText);
+        sendMessage(transcribedText);
+        setIsUserTurn(false);
     }
 
```

Comment:
This is a pretty long method. A comment explaining the high level logic would be helpful